torchgeo/transforms/transforms.py

```diff
@@ -69,3 +74,147 @@ def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
             sample["boxes"] = sample["boxes"].to(boxes_dtype)
 
         return sample
+
+
+class _ExtractTensorPatches(GeometricAugmentationBase2D):
+    """Chop up a tensor into a grid."""
+
+    def __init__(self, window_size: Union[int, Tuple[int, int]]) -> None:
+        """Initialize a new _ExtractTensorPatches instance.
+
+        Args:
+            window_size: the size of each patch
+        """
+        super().__init__(p=1)
+        self.flags = {"window_size": window_size}
+
+    def compute_transformation(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any]
+    ) -> Tensor:
+        """Compute the transformation.
+
+        Args:
+            input: the input tensor
+            params: generated parameters
+            flags: static parameters
+
+        Returns:
+            the transformation
+        """
+        out: Tensor = self.identity_matrix(input)
+        return out
+
+    def apply_transform(
+        self,
+        input: Tensor,
+        params: Dict[str, Tensor],
+        flags: Dict[str, Any],
+        transform: Optional[Tensor] = None,
+    ) -> Tensor:
+        """Apply the transform.
+
+        Args:
+            input: the input tensor
+            params: generated parameters
+            flags: static parameters
+            transform: the geometric transformation tensor
+
+        Returns:
+            the augmented input
+        """
+        size = flags["window_size"]
+        h, w = input.shape[-2:]
+        padding = compute_padding((h, w), size)
+        input = extract_tensor_patches(input, size, size, padding)
+        input = torch.flatten(input, 0, 1)  # [B, N, C?, H, W] -> [B*N, C?, H, W]
+        return input
+
+
+class _RandomNCrop(GeometricAugmentationBase2D):
```

Comment:
I'm concerned that there's a bug in this somewhere. In my tests, if the mask does not have a channel dimension, the crop is incorrect. Can anyone run the trainer and visualize the input/mask for several mini-batches?